Strongbox/Classes/Strongbox.swift

```diff
@@ -103,6 +112,12 @@ public class Strongbox {
     // MARK: Private functions to do all the work
     
     private func set(_ data: NSData?, key: String, accessibility: CFString = kSecAttrAccessibleWhenUnlocked) -> Bool {
+        
+         // The lock prevents the code to be run simultaneously
+         // from multiple threads which may result in crashing
+        lock.lock()
+        defer { lock.unlock() }
```

Comment:
It might make more sense to put this sort of code only in the public functions, rather than in the private functions. Reason being is that you could incur multiple lock/unlock cycles with the lock buried this deep, versus just one cycle when used from the public methods. What do you think?

Reply:
@granoff I agree with your approach. But My suggestion is to avoid changes in the existing project implementation wherever StrongBox.swift is used and avoid writing redundant code blocks or writing a wrapper around StrongBox.Swift to prevent multi-threaded access.
What is you opinion on this Mr. Granoff

Reply:
@granoff I have modified the lock implementation.

Reply:
@granoff Could you please re-review this PR.
Thanks.